tests/route/test_route_consistency.py

```diff
@@ -137,3 +213,150 @@ def test_bgp_shut_noshut(self, duthosts, enum_rand_one_per_hwsku_frontend_hostna
             # startup bgp back in case of any exception
             duthost.shell("sudo config bgp startup all")
             time.sleep(self.sleep_interval)
+
+    @pytest.mark.disable_loganalyzer
+    def test_bgpd_crash_and_recover(self, duthosts):
+        duthost = None
+        for idx, dut in enumerate(duthosts.frontend_nodes):
+            if dut.facts['switch_type'] == "voq" and idx == 0:
+                # pick a UpstreamLC to get higher route churn in VoQ chassis
+                duthost = dut
+        if duthost is None:
+            duthost = duthosts[0]
+        logger.info("test_bgp_crash_and_recover: DUT{}".format(duthost.hostname))
+
+        namespace_ids, succeeded = duthost.get_namespace_ids("bgp")
+        pytest_assert(succeeded, "Failed to get namespace ids of container '{}'".format("bgp"))
+        logger.info("namespace_ids: {}".format(namespace_ids))
+
+        try:
+            logger.info("kill bgpd(s) for {}".format(duthost.hostname))
+            for id in namespace_ids:
+                if id is None:
+                    id = ""
+                check_and_kill_process(duthost, "bgp" + str(id), "bgpd")
+            time.sleep(30)
+
+            post_withdraw_route_snapshot, _ = self.get_route_prefix_snapshot_from_asicdb(duthosts)
+            num_routes_withdrawn = 0
+            for dut_instance_name in self.pre_test_route_snapshot.keys():
+                if num_routes_withdrawn == 0:
+                    num_routes_withdrawn = len(self.pre_test_route_snapshot[dut_instance_name] -
+                                               post_withdraw_route_snapshot[dut_instance_name])
+                    logger.info("num_routes_withdrawn: {}".format(num_routes_withdrawn))
+                else:
+                    assert num_routes_withdrawn == len(self.pre_test_route_snapshot[dut_instance_name] -
+                                                       post_withdraw_route_snapshot[dut_instance_name])
+
+            logger.info("recover bgpd for {}".format(duthost.hostname))
+            config_reload(duthost)
+            time.sleep(self.sleep_interval)
+
+            # take the snapshot of route table from all the DUTs
+            post_test_route_snapshot, _ = self.get_route_prefix_snapshot_from_asicdb(duthosts)
+            for dut_instance_name in self.pre_test_route_snapshot.keys():
+                assert self.pre_test_route_snapshot[dut_instance_name] == post_test_route_snapshot[dut_instance_name]
+            logger.info("Route table is consistent across all the DUTs")
+        except Exception:
+            # startup bgpd back in case of any exception
+            logger.info(traceback.format_exc())
+            config_reload(duthost)
+            time.sleep(self.sleep_interval)
+
+    @pytest.mark.disable_loganalyzer
+    def test_syncd_crash_and_recover(self, duthosts):
```

Comment:
All three tests have essentially the same code, just the process name is changed. Can we club them into one and parametrize on process?